### DISCUSSION TITLE
kafka sizing on GCP

### DISCUSSION THREAD

**Discussion by rishaik05** [strimzi/strimzi-kafka-operator#10347] [Q&A]
team,

Need your help to understand on how to choose node type , max pod and node configuration of kafka on GCP,
Also what kind of file system best suite for strimzi kafka on GCP

**Comment by scholzj** [comment]
Kafka needs block storage (Google Persistent disks) ideally with the xfs filesystem. As for the node types, that mostly depends on your requirements. Kafka typically needs storage and networking, it can make use of memory for caching etc. But whether you need 2CPUs and 16Gi of RAM or 12CPU and 128Gi of RAM will depend on what performance do you need from it. Sometimes, the best way is to do a PoC to see how it behaves in your use-cases with different configurations. There are also many "sizing calculators" online that might give you some initial idea based on some rough numbers in enter into them. If you need availability and reliability, you should also normally need to plan with at least 3 nodes in 3 availability zones and with topics using replication factor 3.
